src/headless/plugins/chat/model.js

```diff
@@ -78,9 +79,24 @@ const ChatBox = ModelWithContact.extend({
          * @example _converse.api.listen.on('chatBoxInitialized', model => { ... });
          */
         await api.trigger('chatBoxInitialized', this, {'Synchronous': true});
+
+        const blocking_supported = await api.blocking?.supported();
+        if (blocking_supported) {
+            await api.waitUntil("blockListFetched");
+            this.checkIfContactBlocked(api.blocking.blocklist());
+          api.listen.on('blockListUpdated', this.checkIfContactBlocked, this);
+        } else this.set({'contact_blocked': undefined});
+
         this.initialized.resolve();
     },
 
+    checkIfContactBlocked (jid_set) {
+        if (jid_set.has(this.get('jid')))
+            return this.set({'contact_blocked': true});
```

Comment:
As a general rule, we always use curly brackets for `if` statements, unless it's a single `if` (no `else`) one a single line.